programs/cleos/main.cpp

```diff
@@ -2958,6 +2958,24 @@ int main( int argc, char** argv ) {
       std::cout << fc::json::to_pretty_string(unpacked_action_data_json) << std::endl;
    });
 
+   string public_key_to_convert;
+   // output public key in legacy format
+   auto public_key_to_legacy = convert->add_subcommand("print_public_key_as_legacy", localized("Output a public key in legacy format (EOS, PUB_R1, PUB_WA)"));
+   public_key_to_legacy->add_option("public_key", public_key_to_convert, localized("Public key to output in legacy format"))->required();
+   public_key_to_legacy->callback([&] {
+      try {
+         std::cout << fc::crypto::public_key(public_key_to_convert).to_legacy_string({}) << std::endl;
+      } EOS_RETHROW_EXCEPTIONS(public_key_type_exception, "Failed to parse public key");
+   });
+   // output public key in new format
+   auto public_key_to_new = convert->add_subcommand("print_public_key_as_new", localized("Output a public key in new format (PUB_K1, PUB_R1, PUB_WA)"));
+   public_key_to_new->add_option("public_key", public_key_to_convert, localized("Public key to output in new format"))->required();
```

Comment:
I wanted to come up with something better then 'new format' but :woman_shrugging:

Reply:
I think `print_` is a bit verbose. Maybe `convert` `public_key_as_legacy`. Seems like print is implied.

Reply:
```
Usage: ./cleos convert [OPTIONS] SUBCOMMAND

Options:
  -h,--help              Print this help message and exit
  --help-all             Show all help


Subcommands:
  pack_transaction       From plain signed JSON to packed form
  unpack_transaction     From packed to plain signed JSON form
  pack_action_data       From JSON action data to packed form
  unpack_action_data     From packed to JSON action data form
```

Existing `convert` is what it is converting to.
To be consistent I think it would be:

```
legacy_public_key       From existing public key to legacy form, i.e. EOS..
public_key              From existing public key to canonical form, i.e. PUB_..
```

Other ideas instead of `canonical`: `new`, `generic`, `standard`, `current`